daemon/src/maker_cfd.rs

```diff
@@ -117,10 +118,34 @@ impl Actor {
         }
     }
 
+    /// Send settlement proposals for the purposes of UI updates.
+    /// Filters out the TakerIds, as they are an implementation detail inside of
+    /// the actor
     fn send_current_settlement_proposals(&self) -> Result<()> {
+        Ok(self.settlements_feed_sender.send(
+            self.current_settlement_proposals
+                .iter()
+                .map(|x| (*x.0, (x.1 .0.clone(), x.1 .1.clone())))
+                .collect(),
+        )?)
+    }
+
+    fn get_taker_id_of_proposal(&self, order_id: &OrderId) -> Result<TakerId> {
         Ok(self
-            .settlements_feed_sender
-            .send(self.current_settlement_proposals.clone())?)
+            .current_settlement_proposals
+            .get_key_value(order_id)
```

Comment:
Why not just `get` if you are throwing away the key anyway?

Reply:
indeed, changed to `get`